Tools/ssl/make_ssl_data.py

```diff
@@ -55,103 +109,178 @@ def _file_search(fname, pat):
+            continue
+        actual = int(num)
+        expect = lib2errnum.setdefault(libname, actual)
+        if actual != expect:
+            logger.warning("OpenSSL inconsistency for ERR_LIB_%s (%d != %d)",
+                           libname, actual, expect)
     return lib2errnum
 
 
 def parse_openssl_error_text(args):
-    """Parse error reasons, X509_R_AKID_MISMATCH"""
-    # ignore backslash line continuation for now
+    """Parse error reasons from crypto/err/openssl.txt.
+
+    Detected lines match "<LIBNAME>_R_<ERRNAME>:<ERRCODE>:<MESSAGE>",
+    e.g., "ASN1_R_ADDING_OBJECT:171:adding object". The <MESSAGE> part
+    is not stored as it will be recovered at runtime when needed.
+    """
+    # ignore backslash line continuation (placed before <MESSAGE> if present)
     pat = re.compile(r"^((\w+?)_R_(\w+)):(\d+):")
-    for match in _file_search(args.errtxt, pat):
+    seen = {}
+    for match in file_search(args.errtxt, pat):
         reason, libname, errname, num = match.groups()
-        if "_F_" in reason:
-            # ignore function codes
+        if "_F_" in reason:  # ignore function codes
+            # FEAT(picnixz): in the future, we may want to also check
+            # the consistency of the OpenSSL files with an external tool.
             continue
-        num = int(num)
-        yield reason, libname, errname, num
+        yield reason, libname, errname, int(num)
 
 
 def parse_extra_reasons(args):
-    """Parse extra reasons from openssl.ec"""
+    """Parse extra reasons from crypto/err/openssl.ec.
+
+    Detected lines are matched against "R <LIBNAME>_R_<ERRNAME> <ERRCODE>",
+    e.g., "R SSL_R_SSLV3_ALERT_UNEXPECTED_MESSAGE 1010".
+    """
     pat = re.compile(r"^R\s+((\w+)_R_(\w+))\s+(\d+)")
-    for match in _file_search(args.errcodes, pat):
+    for match in file_search(args.errcodes, pat):
         reason, libname, errname, num = match.groups()
-        num = int(num)
-        yield reason, libname, errname, num
+        yield reason, libname, errname, int(num)
 
 
 def gen_library_codes(args):
-    """Generate table short libname to numeric code"""
+    """Generate table short libname to numeric code."""
+    yield "/* generated from args.lib2errnum */"
     yield "static struct py_ssl_library_code library_codes[] = {"
     for libname in sorted(args.lib2errnum):
         yield f"#ifdef ERR_LIB_{libname}"
         yield f'    {{"{libname}", ERR_LIB_{libname}}},'
         yield "#endif"
-    yield "    { NULL }"
+    yield "    {NULL, 0}  /* sentinel */"
     yield "};"
-    yield ""
 
 
 def gen_error_codes(args):
-    """Generate error code table for error reasons"""
+    """Generate error code table for error reasons."""
+    yield "/* generated from args.reasons */"
     yield "static struct py_ssl_error_code error_codes[] = {"
     for reason, libname, errname, num in args.reasons:
-        yield f"  #ifdef {reason}"
+        yield f"#ifdef {reason}"
```

Comment:
Could you avoid the style changes?

GitHub hides the diffs in the generated files, but they're the output of a script that's not otherwise tested, so they should be reviewed too. Style changes just make it harder, with no benefit that I can see.

Reply:
Ah, I actually diff with "hide whitespaces" but yes, I can do that.